pkg/hs/king/lib/Ur/King/Main.hs

```diff
@@ -286,6 +290,42 @@ collectAllFx top = do
 
 --------------------------------------------------------------------------------
 
+replayPartEvs :: ∀e. HasLogFunc e => FilePath -> Word64 -> RIO e ()
+replayPartEvs top last = do
+    logTrace $ display $ pack @Text top
+    fetchSnapshot
+    rwith replayedEvs $ \() ->
+        logTrace "Done replaying events!"
+  where
+    fetchSnapshot :: RIO e ()
+    fetchSnapshot = do
+      snap <- Pier.getSnapshot top last
+      case snap of
+        Nothing -> pure ()
+        Just sn -> do
+          let start = top <> "/.partial-replay/"
+          liftIO $ system $ "cp -r \"" <> sn <> "\" \"" <> tmpDir <> "\""
```

Comment:
This is gross and can be broken (file paths maybe contain `"`). However, there doesn't seem to be anything off-shelf for copying directories.

Reply:
`storeSnapshot`?

Reply:
Should probably make sure `.partial-replay` directory exists first.

Reply:
`Pier.getSnapshot` does create `.partial-replay` it it's not there, though it's not obvious that it would do that.

Reply:
Right. It seems wrong to rely on that.

Also, `Pier.getSnapshot` doesn't actually need to do that. It can just handle the case where the directory doesn't exist and return `Nothing`.